src/app/store/features/accounts/accountsSlice.js

```diff
@@ -0,0 +1,29 @@
+const initialState = [
+  { id: 0, text: 'Learn React', completed: true },
+  { id: 1, text: 'Learn Redux', completed: false, color: 'purple' },
+  { id: 2, text: 'Build something fun!', completed: false, color: 'blue' }
+]
+
+function nextAccountId(accounts) {
+  const maxId = accounts.reduce((maxId, account) => Math.max(account.id, maxId), -1)
+  return maxId + 1
+}
+
+
+export default function accountsReducer(state = initialState, action) {
+  switch (action.type) {
+    case 'accounts/addAccount': {
+      // Can return just the new accounts array - no extra object around it
```

Comment:
would I implement certain business logic here, too? or is this only for data storage?

Reply:
https://medium.com/@jeffbski/where-do-i-put-my-business-logic-in-a-react-redux-application-9253ef91ce1

Reply:
which option is your favorit? joule used sagas.

Reply:
without much refactoring we'd be at "Dispatching from action creators", meaning we'd call the store to inform it about new data after we've invoked a connector call for example. I'd very much vote for this approach as much as possible in  a single PR - we'd be able to refactor introduce new stuff with PR's further down the road.